### ISSUE TITLE
JS error in role details view

### ISSUE THREAD

**Issue by digedag** [orocrm/platform#584]
I registered an new privilege category provider for my entities. After that it is not possible to view or edit a role entity any more. The browser will raise a JavaScript error: **TypeError: categories.unshift is not a function** in **entity-category-tabs-component.js**.

There is an array expected but an object given. The PHP array used as input in tabsOptions.data looks like this:
```php
Array
(
    [0] => Array
        (
            [id] => account_management
            [label] => Account Management
        )

    [1] => Array
        (
            [id] => marketing
            [label] => Marketing
        )

    [2] => Array
        (
            [id] => sales_data
            [label] => Verkaufsdaten
        )

    [5] => Array
        (
            [id] => my_data
            [label] => My Bundle Entities
        )
)
```
The index value skips from 2 to 5, so the array is associative and json_encode will build an JS-object instead of JS-array.

**Comment by mkudelya** [comment]
Hi @digedag 

How can i reproduce the issue ?

**Comment by digedag** [comment]
I want to add a new tab in role view for my bundles entities. The screenshot shows the result with my fix.

![view_role_tabs](https://cloud.githubusercontent.com/assets/965877/21310046/01b7ff08-c5e1-11e6-9b71-d7960adcbea3.png)

So I registered a new PrivilegeCategoryProvider.
```php
class PrivilegeCategoryProvider implements PrivilegeCategoryProviderInterface
{
    const NAME = 'my_data';

    /**
     * {@inheritdoc}
     */
    public function getName()
    {
        return self::NAME;
    }

    /**
     * {@inheritdoc}
     */
    public function getRolePrivilegeCategory()
    {
        return new PrivilegeCategory(self::NAME, 'mybundle.basics.privilege.category.my_data.label', true, 20);
    }
}
```
and in services.yml:
```yaml
    my_basics.provider.privilege_category_provider:
        class: "%my_basics.provider.privilege_category_provider.class%"
        tags:
            - { name: oro_user.privilege_category}
```
Finally I added my entites to security category my_data:
```
 * @Config(
 *   "security"={
 *     "type"="ACL",
 *     "group_name"="",
 *     "category"="my_data"
 *   },
```
Under user/role/view/ all of my entites should be grouped in a new tab "my data". But instead I get the JS error.

**Comment by mkudelya** [comment]
I have tested your case on 1.10.12 and i didn't get any errors. Which version of orocrm are you using ? Did you wrap 'security' section in defaultValues ?

**Comment by digedag** [comment]
I'm running also on 1.10.12 with PHP 5.6 as well as with PHP 7. And the defaultValues section wraps my security part in entity annotation.
Now I did some more investigation to reproduce your successful test. So I checked what are the filtered entries from getPermissionCategories() in [RolePrivilegeCategoryProvider](https://github.com/digedag/platform/blob/03ed7e933f98b6adae43fb09d3f953049f5150f0/src/Oro/Bundle/UserBundle/Provider/RolePrivilegeCategoryProvider.php#L153). The complete array from my first comment lookes like this:
```php
Array
        (
            [0] => Oro\Bundle\UserBundle\Model\PrivilegeCategory Object
                (
                    [id:protected] => account_management
                    [label:protected] => oro.user.privilege.category.account_management.label
                    [tab:protected] => 1
                    [visible:protected] => 1
                    [priority:protected] => 0
                )

            [1] => Oro\Bundle\UserBundle\Model\PrivilegeCategory Object
                (
                    [id:protected] => marketing
                    [label:protected] => orocrm.campaign.privilege.category.marketing.label
                    [tab:protected] => 1
                    [visible:protected] => 1
                    [priority:protected] => 5
                )

            [2] => Oro\Bundle\UserBundle\Model\PrivilegeCategory Object
                (
                    [id:protected] => sales_data
                    [label:protected] => orocrm.sales.privilege.category.sales_data.label
                    [tab:protected] => 1
                    [visible:protected] => 1
                    [priority:protected] => 7
                )

            [3] => Oro\Bundle\UserBundle\Model\PrivilegeCategory Object
                (
                    [id:protected] => address
                    [label:protected] => oro.user.privilege.category.address.label
                    [tab:protected] => 
                    [visible:protected] => 1
                    [priority:protected] => 10
                )

            [4] => Oro\Bundle\UserBundle\Model\PrivilegeCategory Object
                (
                    [id:protected] => application
                    [label:protected] => oro.user.privilege.category.application.label
                    [tab:protected] => 
                    [visible:protected] => 1
                    [priority:protected] => 20
                )

            [5] => Oro\Bundle\UserBundle\Model\PrivilegeCategory Object
                (
                    [id:protected] => my_data
                    [label:protected] => acme.mybundle.privilege.category.my_data.label
                    [tab:protected] => 1
                    [visible:protected] => 1
                    [priority:protected] => 20
                )

            [6] => Oro\Bundle\UserBundle\Model\PrivilegeCategory Object
                (
                    [id:protected] => calendar
                    [label:protected] => oro.user.privilege.category.calendar.label
                    [tab:protected] => 
                    [visible:protected] => 1
                    [priority:protected] => 30
                )

            [7] => Oro\Bundle\UserBundle\Model\PrivilegeCategory Object
                (
                    [id:protected] => entity
                    [label:protected] => oro.user.privilege.category.entity.label
                    [tab:protected] => 
                    [visible:protected] => 1
                    [priority:protected] => 40
                )

        )
```
As you can see, the data is ordered by priority. My Provider has priority 20 on index 5. The providers on index 3 and 4 have no tabs. So array_filter will return an associative array. As a workaround I can raise priority to a value lower then 10. This works. But it will break again if someone registers a new provider without tabs with a priority 1.

If you use array_filter() and you expect the result has to be an array after json_encode() [you have to take care about the keys](http://php.net/manual/de/function.array-filter.php#99358).

**Comment by mkudelya** [comment]
Hi @digedag 

Thank you for your contribution. Your pull request will be available in 1.10 and master branches as soon as possible.
